src/Agent.Plugins/PipelineArtifact/FilePathUNCServer.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Agent.Sdk;
+using Microsoft.TeamFoundation.Build.WebApi;
+using Microsoft.VisualStudio.Services.WebApi;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Microsoft.VisualStudio.Services.Content.Common.Tracing;
+
+namespace Agent.Plugins.PipelineArtifact
+{
+    public class FilePathUNCServer
+    {
+        internal async Task UploadAsync(
+            AgentTaskPluginExecutionContext context,
+            Guid projectId,
+            int buildId,
+            string artifactName,
+            string targetPath,
+            string fileSharePath,
+            CancellationToken token)
+        {
+            string artifactPath = Path.Join(fileSharePath, artifactName);
```

Comment:
>fileSharePath [](start = 44, length = 13)

We should definitely do some UNC path validation to ensure it is indeed a UNC path

Reply:
Pipeline artifact and fileShare is using the same UI and the path is already checked/verified in the PipelineArtifactPlugin.